types/src/vm/module_metadata.rs

```diff
@@ -740,3 +719,65 @@ impl RandomnessAnnotation {
         Self { max_gas }
     }
 }
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum ResourceGroupScope {
+    Global,
+    Address,
+    Module,
+}
+
+impl ResourceGroupScope {
+    pub fn is_less_strict(&self, other: &ResourceGroupScope) -> bool {
```

Comment:
let's add one unit test for this and for are_equal_module_ids, also confirming that the logic makes sense?